trie/trienode/node.go

```diff
@@ -42,35 +42,13 @@ func (n *Node) IsDeleted() bool {
 	return n.Hash == (common.Hash{})
 }
 
-// WithPrev wraps the Node with the previous node value attached.
-type WithPrev struct {
-	*Node
-	Prev []byte // Encoded original value, nil means it's non-existent
-}
-
-// Unwrap returns the internal Node object.
-func (n *WithPrev) Unwrap() *Node {
-	return n.Node
-}
-
-// Size returns the total memory size used by this node. It overloads
-// the function in Node by counting the size of previous value as well.
-func (n *WithPrev) Size() int {
-	return n.Node.Size() + len(n.Prev)
-}
-
 // New constructs a node with provided node information.
 func New(hash common.Hash, blob []byte) *Node {
 	return &Node{Hash: hash, Blob: blob}
 }
 
-// NewWithPrev constructs a node with provided node information.
-func NewWithPrev(hash common.Hash, blob []byte, prev []byte) *WithPrev {
-	return &WithPrev{
-		Node: New(hash, blob),
-		Prev: prev,
-	}
-}
+// NewDeleted constructs a node which is deleted.
+func NewDeleted() *Node { return New(common.Hash{}, nil) }
```

Comment:
Should perhaps the hash be the empty hash and not the zero hash?

Reply:
It's just a flag for deletion marker, zero-hash and nil data all are special flags for it.